exla/c_src/exla/exla_cuda.cc

```diff
@@ -5,24 +5,31 @@
 
 #include <cstring>
 #include <iostream>
+#include <optional>
+#include <string>
 
-std::pair<std::vector<unsigned char>, int> get_cuda_ipc_handle(std::uintptr_t ptr) {
+std::optional<std::string> get_cuda_ipc_handle(std::uintptr_t ptr) {
   cudaIpcMemHandle_t ipc_handle;
   cudaError_t status = cudaIpcGetMemHandle(&ipc_handle, reinterpret_cast<void*>(ptr));
 
+  if (status != cudaSuccess) {
+    return std::nullopt;
+  }
+
   // Assuming sizeof(cudaIpcMemHandle_t) is constant
   const size_t size = sizeof(cudaIpcMemHandle_t);
 
-  // Copy the memory handle to a byte array
-  std::vector<unsigned char> result(size);
-  memcpy(result.data(), &ipc_handle, size);
+  // Copy the memory handle to a buffer
+  std::string buffer;
+  buffer.resize(size);
+  memcpy(&(*(buffer.begin())), &ipc_handle, size);
 
-  return std::make_pair(result, status != cudaSuccess);
+  return buffer;
 }
 
```

Comment:
I don't think we should be using std::string here

Reply:
`std::string` is actually just a sequence of `char`, the data is not said to have any encoding or similar, so it's basically Elixir binary. The buffer has 1 additional \0 byte at the end, so we can get `string.c_str()`, but if we use string as a buffer, this doesn't matter.

One thing though is that I did `&(*(buffer.begin()))` here, as it was elsewhere, but I later realised we should not do that (it works, but it is not the right thing to do). I will update it in the next PR.

Or is there any other reason not to use `std::string`?

Reply:
The practical reason is that using a string, encoding is done automatically, otherwise we would need to convert the vector to binary term by hand (which is fine, but my point is that either of them is just buffer) :D

And for `host_ipc` the handle is an "actual" string, but in both cases we tread it as an opaque binary.